lib/puppet/provider/zone/solaris.rb

```diff
@@ -37,43 +53,42 @@ def self.instances
   # file or string.
   def configure
 
-    if @resource[:archive].nil? && @resource[:zonecfg_export].nil?
+    if @resource[:zonecfg_archive].nil? && @resource[:zonecfg_export].nil?
       raise Puppet::Error, "No configuration resource is defined."
     end
 
-    command = String.new
+    command = "#{command(:cfg)} -z #{@resource[:name]}"
     if @resource[:clone]
       if !@resource[:zonecfg_export]
         raise Puppet::Error, "A zone configuration must be defined to
         clone a zone."
       end
-      command = "#{command(:cfg)} -z #{@resource[:name]} -f #{@resource[:zonecfg_export]}"
+      command << " -f #{@resource[:zonecfg_export]}"
```

Comment:
Why does clone use -f to load the file and non-clone configs read the file or parameter into a string?